### PR TITLE
[quant][graphmode] Add support for reference pattern for torch.cat

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62376
* #62375
* __->__ #62374
* #62348

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_cat

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29979713](https://our.internmc.facebook.com/intern/diff/D29979713)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29979713).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29979713).

File: torch/quantization/fx/quantization_patterns.py [hunk 1]
```diff
@@ -445,7 +445,29 @@ def convert(self,
                 convert_custom_config_dict: Dict[str, Any] = None) -> Node:
         if not self.all_node_args_are_tensors:
             return NotImplemented
-        return quantized_graph.node_copy(node, load_arg(quantized=torch.quint8))
+        if is_reference:
+            activation_post_process = \
+                self._maybe_get_last_node_only_observer(modules)
+            if activation_post_process is None:
```
Comment (vkuzo) [inline torch/quantization/fx/quantization_patterns.py:451]: just curious, what's the context for this check? is this checking if the op is in floating point (so reference pattern is equivalent to the op)?  If yes, would it make sense to check the expected dtype instead?

**Review by vkuzo — APPROVED** [review]

File: torch/quantization/fx/quantization_patterns.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:451]: oh this is for cases like dynamic quantization and weight only quantization where no observer is required for the input and output activations. however dynamic quantization would quantize the activation internally so the numerics is still not matching for that case. right, for this case I think the dtype for activation should be torch.float, I can add the check

Comment (jerryzh168) [reply torch/quantization/fx/quantization_patterns.py:451]: will add this in a future PR

**Pull request closed without merging**